app/controllers/admin/project_steps_controller.rb

```diff
@@ -123,6 +123,9 @@ def duplicate
   end
 
   def batch_destroy
+    # don't initialize batch destroy if there are no step ids
```

Comment:
The UI should catch any cases when there are no steps selected. Are the changes in this file just a safeguard?

Reply:
Yep
The UI would catch those due to the frontend manipulation but this safeguards our backend and makes sure no dirty data gets in, especially the wrong way